api/entity_operations.go

```diff
@@ -61,25 +172,75 @@ func (c *Client) AddOrgEntity(transaction map[string]interface{}, entityCounters
 	entityCounter := entityCounters[childType] + 1
 	newEntityID := fmt.Sprintf("%s_%d", prefix, entityCounter)
 
-	// Get the parent entity ID
-	searchCriteria := &models.SearchCriteria{
-		Kind: &models.Kind{
-			Major: "Organisation",
-			Minor: parentType,
-		},
-		Name: parent,
-	}
+	// Get the parent entity ID based on the child type
+	var parentID string
 
-	searchResults, err := c.SearchEntities(searchCriteria)
-	if err != nil {
-		return 0, fmt.Errorf("failed to search for parent entity: %w", err)
-	}
+	if childType == "minister" {
```

Comment:
same here. All string constant values need to refer to an enum or a constant variable the least.